packages/core/src/DirectoryLoader.ts

```diff
@@ -54,6 +57,58 @@ export abstract class DirectoryLoader {
 		return path.resolve(this.directory, file);
 	}
 
+	protected isOAuth(credType: ICredentialType) {
+		return (
+			Array.isArray(credType.extends) &&
+			credType.extends.some((parentType) =>
+				['oAuth2Api', 'googleOAuth2Api', 'oAuth1Api'].includes(parentType),
+			)
+		);
+	}
+
+	/**
+	 * Whether any of the node's credential types may be used to
+	 * make a request from a node other than itself.
+	 */
+	protected supportsProxyAuth(description: INodeTypeDescription) {
+		if (!description.credentials) return false;
+
+		// const credentialTypes = CredentialTypes();
+
+		return description.credentials.some(({ name }) => {
+			const credType = this.credentialTypes[name].type;
+
+			if (credType.authenticate !== undefined) return true;
+
+			return this.isOAuth(credType);
+		});
+	}
+
+	/**
+	 * Inject a `Custom API Call` option into `resource` and `operation`
+	 * parameters in a node that supports proxy auth.
+	 */
+	protected injectCustomApiCallOption(description: INodeTypeDescription) {
+		if (!this.supportsProxyAuth(description)) return description;
+
+		description.properties.forEach((p) => {
+			if (
+				['resource', 'operation'].includes(p.name) &&
+				Array.isArray(p.options) &&
+				p.options[p.options.length - 1].name !== CUSTOM_API_CALL_NAME
+			) {
+				p.options.push({
+					name: CUSTOM_API_CALL_NAME,
+					value: CUSTOM_API_CALL_KEY,
+				});
+			}
+
+			return p;
+		});
+
+		return description;
```

Comment:
Do we need to return the description if we are only relying on the mutation? Same for L92.